sql/moz-fx-data-shared-prod/analysis/unified_metrics_v1/query.sql

```diff
@@ -0,0 +1,258 @@
+WITH unioned_source AS (
+  SELECT
+    *,
+    CAST(NULL AS string) AS distribution_id,
+    'Fenix' AS normalized_app_name
+  FROM
+    fenix.clients_last_seen_joined
+  WHERE
+    submission_date = @submission_date
+  UNION ALL
+  SELECT
+    *,
+    CAST(NULL AS string) AS distribution_id,
+    'Firefox iOS' AS normalized_app_name
+  FROM
+    firefox_ios.clients_last_seen_joined
+  WHERE
+    submission_date = @submission_date
+  UNION ALL
+  SELECT
+    *,
+    CAST(NULL AS string) AS distribution_id,
+    'Focus iOS' AS normalized_app_name
+  FROM
+    focus_ios.clients_last_seen_joined
+  WHERE
+    submission_date = @submission_date
+  UNION ALL
+  SELECT
+    submission_date,
+    normalized_channel,
+    client_id,
+    udf_js.sample_id(client_id) AS sample_id,
+    days_since_seen,
+    days_since_created_profile,
+    NULL AS days_since_seen_session_start,
+    NULL AS days_since_seen_session_end,
+    days_seen_bits,
+    days_created_profile_bits,
+    first_seen_date AS first_run_date,
+    durations,
+    NULL AS days_seen_session_start_bits,
+    NULL AS days_seen_session_end_bits,
+    os AS normalized_os,
+    osversion AS normalized_os_version,
+    NULL AS android_sdk_version,
+    locale,
+    city,
+    country,
+    app_build_id AS app_build,
+    NULL AS app_channel,
+    metadata_app_version AS app_display_version,
+    NULL AS architecture,
+    NULL AS device_manufacturer,
+    device AS device_model,
+    NULL AS telemetry_sdk_build,
+    first_seen_date,
+    submission_date = first_seen_date AS is_new_profile,
+    NULL AS n_metrics_ping,
+    NULL AS days_sent_metrics_ping_bits,
+    NULL AS uri_count,
+    default_browser AS is_default_browser,
+    distribution_id AS distribution_id,
+    'Focus Android' AS normalized_app_name,
+    NULL AS isp
+  FROM
+    telemetry.core_clients_last_seen
+  WHERE
+    submission_date = @submission_date
+    AND app_name = 'Focus'
+    AND os = 'Android'
+),
+unioned AS (
+  SELECT
+    * REPLACE (
+      -- Per bug 1757216 we need to exclude BrowserStack clients from KPIs,
+      -- so we mark them with a separate app name here. We expect BrowserStack
+      -- clients only on release channel of Fenix, so the only variant this is
+      -- expected to produce is 'Fenix BrowserStack'
+      IF(
+        isp = 'BrowserStack',
+        CONCAT(normalized_app_name, ' BrowserStack'),
+        normalized_app_name
+      ) AS normalized_app_name
+    )
+  FROM
+    unioned_source
+),
+search_clients AS (
+  SELECT
+    *
```

Comment:
Do we actually need all columns here? I assume this table will be using clustered fields and if only a few columns are needed could help improve performance.

Reply:
For now we decided not to remove fields as this table is used by more people.